oracle-uri-proposal.md

```diff
@@ -0,0 +1,197 @@
+
+A single URI string that uniquely identifies an event has the following benefits:
+
+1. Copy pasteable: A user can easily copy paste the event id into a chat app/DLC software/browser address bar.
+2. Trivially serializable: It is easy to byte encode the event id for signing purposes.
+3. Human readable: A user who is sufficiently familiar with event namespace does not need software to understand the event.
+4. Portable: If two oracles support the same event id then you can use either of the oracles without having to query the oracles to determine if they are attesting to the same event, and just as importantly, in the same way.
+
+## Specification
+
+An event identifier (_event id_) is a relative URI reference (a URI without the scheme and authority) with a query string.
+The path of the URI is interpreted as a description of an event in time and space.
+The query string is interpreted as an _measurement_ and describes a certain fact that will be observed by an oracle on the object described by the path.
+
+If applicable, the path should start with the human organization responsible for the event.
+If it is a price then the name of the exchange where the price is being measured e.g. `binance` or if it's a sporting competition then the name of the competition e.g. `NBA`.
+If there is no applicable organization because the measurement is being taken on some natural phenomenon like the weather or the passage of time then the first segment of the path should just be the name of this phenomenon e.g. `weather` or `time`.
+
+Each following segment, should narrow the event id until it focuses on a particular event in space and time. 
+Oracles are free to define the structure for this themselves with the following considerations:
+
+1. The event type of the event may require the path has a certain structure.
+2. Rules and conventions may be attached to namespaces and in general oracles should try and follow them.
+
+### Announcement
+
+Oracles must announce their intention to attest to the outcome of an event ahead of time.
+This is done by signing a message which includes the event id.
+Additionally it usually involves committing to some auxiliary cryptographic data (e.g. nonces) which allows the clients to _anticipate_ the final signature for any message.
+The announcement signature is made by suffixing the event id with `!` and appending the cryptographic data.
+
+_how to encode the data is left for a future revision_
+
+### Attestation
+
+When attesting to a particular outcome for the event the oracle appends `=` and the outcome text onto the event id and signs the result.
+For example, if the outcome of `NBA/s2019/2020-08-11/PHI_PHX.vs` is `PHX_win` then the oracle signs `NBA/s2019/2020-08-11/PHI_PHX.vs=PHX_win`.
+
+### Source Identifiers
+
+If the first segment of a path contains an `@` symbol then the text that precedes it is a _source identifier_.
```

Comment:
Yeah it might depend on implementations, but from my understanding of [the rfc](https://tools.ietf.org/html/rfc3986#section-2.2) if you want to use "@" it needs to be percentage encoded.

Reply:
It also might be annoying on twitter. Perhaps the url can just have a fragment and after the `#` we just serialize metadata like this that doesn't affect the outcomes in some way. So it doesn't appear in the URL but is signed in the announcement message.